app/components/Search/SearchContainer.js

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Body,
+  Icon,
+  Input,
+  Item,
+  Left,
+  ListItem,
+  Text,
+  View,
+} from 'native-base';
+import { StyleSheet, TouchableOpacity } from 'react-native';
+import theme from '../../theme';
+import Autocomplete from 'native-base-autocomplete';
+import { useDispatch, useSelector } from 'react-redux';
+import * as stockActions from '../Stock/store/actions';
+import * as searchActions from './store/actions';
+import * as watchlistActions from '../Watchlist/store/actions';
+import { Actions as RouteActions } from 'react-native-router-flux';
+import { includes } from 'lodash';
+
+function SearchContainer() {
+  const [query, setQuery] = useState('');
+  const results = useSelector(({ search }) => search.data.results);
+  const symbols = useSelector(({ watchlist }) => watchlist.symbols);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(searchActions.getSymbols());
+  }, []);
+
+  useEffect(() => {
+    dispatch(searchActions.getResults(query));
+  }, [query]);
+
+  const handleSelectItem = item => {
+    dispatch(stockActions.setSymbol(item));
+    RouteActions.stockScreen();
+  };
+
+  const handleSearch = text => {
+    const q = text.trim();
+
+    setQuery(q);
+  };
+
+  const handleAddToWatchlist = symbol => {
+    return dispatch(watchlistActions.addWatchlistSymbol(symbol));
+  };
+
+  const handleWatchlistIconStyles = symbol => {
+    return includes(symbols, symbol) ? styles.negative : styles.positive;
+  };
+
+  const handleHideResults = results.length < 1 || query === '';
```

Comment:
it is enough to check that the query is an empty string for hiding the results.Also you already made a condition in the reducer which return an empty string if the query is empty

Reply:
Where do you think is better to handle the hideResults condition?  In the reducer or in the component itself?  I'm thinking the reducer would be better so that I'm not storing results in the application state that never will be rendered in the component.

Reply:
moved to reducer.  
resolved 9e8776e55d69ce440e694e921206dc476c632482